smithy-model/src/main/java/software/amazon/smithy/model/validation/validators/PaginatedTraitValidator.java

```diff
@@ -120,20 +122,26 @@
             PropertyValidator validator
     ) {
         String prefix = service != null ? "When bound within the `" + service.getId() + "` service, " : "";
-        String memberName = validator.getMemberName(opIndex, operation, trait).orElse(null);
+        String memberPath = validator.getMemberPath(opIndex, operation, trait).orElse(null);
 
-        if (memberName == null) {
+        if (memberPath == null) {
             return service != null && validator.isRequiredToBePresent()
                    ? Collections.singletonList(error(operation, trait, String.format(
                     "%spaginated trait `%s` is not configured", prefix, validator.propertyName())))
                    : Collections.emptyList();
         }
 
-        MemberShape member = validator.getMember(opIndex, operation, trait).orElse(null);
+        if (!validator.pathsAllowed() && memberPath.split("\\.").length != 1) {
```

Comment:
Minor: Storing the regex in a constant means we don't need to re-compile it each time. Could you just use `memberPath.contains(".")`?